sys/timer.c

```diff
@@ -0,0 +1,113 @@
+#define KL_LOG KL_TIME
+#include <timer.h>
+#include <mutex.h>
+#include <errno.h>
+#include <klog.h>
+#include <interrupt.h>
+
+static mtx_t timers_mtx = MTX_INITIALIZER(MTX_DEF);
+static timer_list_t timers = TAILQ_HEAD_INITIALIZER(timers);
+
+#define TMF_ACTIVE 0x1000
+#define TMF_INITIALIZED 0x2000
+#define TMF_ALLOCATED 0x4000
+#define TMF_REGISTERED 0x8000
+
```

Comment:
Please explain in a comment that above defined flags are for internal use and need not be seen outside the file.

Reply:
Done!